src/chunker/rabin.js

```diff
@@ -30,30 +28,216 @@ module.exports = async function * rabinChunker (source, options) {
   }
 
   const sizepow = Math.floor(Math.log2(avg))
-  const rabin = createRabin({
+
+  for await (const chunk of rabin(source, {
     min: min,
     max: max,
     bits: sizepow,
     window: options.window,
     polynomial: options.polynomial
-  })
+  })) {
+    yield chunk
+  }
+}
+
+const nativeRabin = () => {
+  const createRabin = require('rabin')
+
+  if (typeof rabin !== 'function') {
+    throw errCode(new Error(`rabin was not a function`), 'ERR_UNSUPPORTED')
+  }
+
+  return async function * (source, options) {
+    const rabin = createRabin(options)
+
+    // TODO: rewrite rabin using node streams v3
+    for await (const chunk of source) {
+      rabin.buffers.append(chunk)
+      rabin.pending.push(chunk)
+
+      const sizes = []
+
+      rabin.rabin.fingerprint(rabin.pending, sizes)
+      rabin.pending = []
+
+      for (let i = 0; i < sizes.length; i++) {
+        const size = sizes[i]
+        const buf = rabin.buffers.slice(0, size)
+        rabin.buffers.consume(size)
+
+        yield buf
+      }
+    }
+
+    if (rabin.buffers.length) {
+      yield rabin.buffers.slice(0)
+    }
+  }
+}
+
+const jsRabin = () => {
+  // see https://github.com/datproject/rabin/blob/c0378395dc0a125ab21ac176ec504f9995b34e62/src/rabin.cc
+  class Rabin {
```

Comment:
`npm i rabin.js`?

Reply:
Do you mean [rabinjs](https://www.npmjs.com/package/rabinjs)?  It has no repo on npm and only 5 downloads a week.

Not sure it's ready for production.